session/pingpong/accountant_promise_settler.go

```diff
@@ -267,13 +319,23 @@ func (aps *AccountantPromiseSettler) settle(p receivedPromise) {
 		defer aps.setSettling(p.provider, false)
 		select {
 		case <-aps.stop:
+			// send a signal to waiter that the settlement was cancelled
+			if waiter != nil {
```

Comment:
Strange conditions here, can waiter be nilable? Maybe different function or nop waiter